### ISSUE TITLE
Bug in azure-streamanalytics-cicd causes our azure pipelines to fail

### ISSUE THREAD

**Issue by joost-lambregts** [Azure/azure-stream-analytics#171]
The azure-streamanalytics-cicd npm package downloads a zipfile. The zipfile downloaded depends on the installed version of .net SDK on the machine. The code is locaded in install.js. This is the relevant code snippet:

```javascript
// Check .Net SDK version for VS project.
exec(`dotnet --list-sdks`, (error, stdout, stderr) => {
    try {
        if (!!stdout) {
            const reg = new RegExp('[36].[0-9.]+', 'g');
            const matches = stdout.match(reg);
            if (matches) {
                dotnetVersion = matches[matches.length - 1].match(/[36].[0-9.]+/)[0];
                if (compareVersions(dotnetVersion, '3.1.0') >= 0) {
                    console.log('.NET Core environment is ready.');
                    resolve(dotnetVersion);
                }
            }
        }
    }
```

The bug is in the regex. '.' is a reserved character and needs to be escaped. As it is, the regex matches 302 in 7.0.302. In our case, this is the last version outputted by dotnet --list-sdks, and so the dotnetVersion variable is set to 7.0.302. 
Later this then causes the code to default to 3.x instead of 6.x:
```javascript
const fileList = JSON.parse(data.toString());
            // default version is 3.x
            let endpoint = fileList[0].fileUrl;

            switch(dotnetVersion[0]){
                case '6':
                    endpoint = fileList[1].fileUrl;
                    break;
                case '3':
                default:
                    endpoint = fileList[0].fileUrl;
                    break;
            }
```

I honestly feel like there should not be a default at all. If you fall into the default branch, that means that you did not find an installed 3.x or 6.x version. At that point you should just throw an error. What happens now is that the code just continues and downloads the 3.x version of the zip, and then later we encounter an error because we don't have .net 3.x installed. That put us on the wrong track when debugging the issue.

**Comment by joost-lambregts** [comment]
Another problem with the regex is that is would identify any version of the form x.3.y as a 3.x version, and any version of the form x.6.y as a 6.x version. 
The regex should just be this (though technically you get into trouble if you ever get to version 13):
[36]\.\d+\.\d+
so the javascript code should be this: (need to escape the backslashes)
`const reg = new RegExp('[36]\\.\\d+\\.\\d+', 'g');`

**Comment by ruiminwang** [comment]
Hi @joost-lambregts , sorry for breaking your pipeline because of this bug and thank you for bringing up this issue.
This issue also broke our test pipeline recently, we have fixed it but haven't release the hotfix yet. Will publish asap.
To work around temporarily, you could install .Netcore 3.1 to your pipeline agent.

**Comment by joost-lambregts** [comment]
Hi @ruiminwang , thank you for the update. We have found another short term workaround, we are using an older runner image that still has .Netcore 3.1 installed. This is not ideal of course, so I'm looking forward to the fix.

**Comment by ruiminwang** [comment]
Hi @joost-lambregts , I just published 3.1.1 for hotfix, could you please have a try?

**Comment by joost-lambregts** [comment]
@ruiminwang It works!
